lncore/peers.go

```diff
@@ -1,21 +1,17 @@
 package lncore
 
-// LnAddr is just a bech32-encoded pubkey.
-// TODO Move this to another package so it's more obviously not *just* IO-related.
-type LnAddr string
-
 // LitPeerStorage is storage for peer data.
 type LitPeerStorage interface {
-	GetPeerAddrs() ([]LnAddr, error)
-	GetPeerInfo(addr LnAddr) (*PeerInfo, error)
-	GetPeerInfos() (map[LnAddr]PeerInfo, error)
-	AddPeer(addr LnAddr, pi PeerInfo) error // also updates
-	DeletePeer(addr LnAddr) error
+	GetPeerAddrs() ([]string, error)
+	GetPeerInfo(addr string) (PeerInfo, error)
+	GetPeerInfos() (map[string]PeerInfo, error)
+	AddPeer(addr string, pi PeerInfo) error // also updates
+	DeletePeer(addr string) error
 }
 
-// PeerInfo .
+// PeerInfo stores peer related data
 type PeerInfo struct {
-	LnAddr   *LnAddr `json:"lnaddr"`
-	Nickname *string `json:"name"`
-	NetAddr  *string `json:"netaddr"` // ip address, port, I guess
+	Addr     string `json:"string"`
```

Comment:
Why get rid of `LnAddr`?

Reply:
We must define the stuff in the function definition as `addr string` instead of defining a type `LnAddr` and then calling `addr LnAddr`. Also, I don't see the type of the address (string) changing for now, so I don't see why we should have a redundant type when we just can use `string`.

Reply:
Really it makes sense to use `string` externally but then internally use `LnAddr`.  With the idea that we're enforcing if the type is a `LnAddr` then it *must* be a valid address and if it's just a `string` then we haven't verified that yet.  Also it being a different type stops people from directly modifying it without trying.

Reply:
If the type is a string, it could be an address (valid or not). There should be appropriate functions to check if its valid or not and that mustn't move on to the types themselves (since that encourages bad practices wrt code quality). Something like `string.IsValidAdr()` would work.

Reg. people using it, I do agree we need a schema of sorts to prevent people from changing stuff. If people can change `addr string` to `addr blah`, they could change `addr LnAddr` to `addr blah` equally easily. I'm saying that we need to fix the root problem, which is not having handlers for all types that we use and that a new type set doesn't change that. I'll put the handlers in a follow-up PR once this is merged :)

Reply:
>that encourages bad practices wrt code quality

I beg to differ.  I think that actually encourages good code quality because as long as you don't intentionally break invariants you can rely on the types to encode the semantics of the data (very common in pure-FP languages).  It also "hides" behavior (like passing a `LnAddr` to things in the `strings` package) that you don't need anymore and stops you from doing "bad" things.

Granted, there's nothing stopping you from doing `x := lncore.LnAddr(strings.SomethingBad(string(x)))` but there's nothing we can do about that.